okhttp-logging-interceptor/src/main/java/com/squareup/okhttp/logging/HttpLoggingInterceptor.java

```diff
@@ -179,7 +179,7 @@ public Level getLevel() {
       }
 
       String endMessage = "--> END " + request.method();
-      if (logBody && hasRequestBody) {
+      if (logBody && hasRequestBody && logBody(request.headers())) {
```

Comment:
Fixed. I also restructured this a bit so we can suppress logging for non-text bodies. (That's a separate PR.)